### PR TITLE
PTNFLY-906 TableView pagination

### PR DESCRIPTION
**Pull request by dlabrecq**
## Description
Added pagination functionality to TableView. 

Created a new Datatables.net extension, named pfPagination. The JS file is optionally included in the page with our other extensions for filtering, row selection, etc.

### Jira
https://patternfly.atlassian.net/browse/PTNFLY-906

### Examples
https://rawgit.com/dlabrecq/patternfly/PTNFLY-906-dist/dist/tests/table-view.html
https://rawgit.com/dlabrecq/patternfly/PTNFLY-906-dist/dist/tests/table-view-navbar.html

screen shot 2017-01-27 at 11 37 36 pm

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi, I noticed if I set a filter which returns 0 results, for some reason the 'select all' checkbox becomes checked.

**Comment by dtaylor113** [comment]
LGTM!  Nice coding style :-)

File: src/js/patternfly.dataTables.pfPagination.js [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+    ctx._pfPagination.currentItems = $(CURRENT_ITEMS_SELECTOR, opts.paginationSelector); // Current items
+    ctx._pfPagination.currentPage = $(CURRENT_PAGE_SELECTOR, opts.paginationSelector); // Current page
+    ctx._pfPagination.firstPage = $(FIRST_PAGE_SELECTOR, opts.paginationSelector); // First page button
+    ctx._pfPagination.forwardActions = $(FORWARD_ACTIONS_SELECTOR, opts.paginationSelector); // Forward navigation actions
+    ctx._pfPagination.lastPage = $(LAST_PAGE_SELECTOR, opts.paginationSelector); // Last page button
+    ctx._pfPagination.nextPage = $(NEXT_PAGE_SELECTOR, opts.paginationSelector); // Next page button
+    ctx._pfPagination.pageSize = $(PAGE_SIZE_SELECTOR, opts.paginationSelector); // Page size selection
+    ctx._pfPagination.previousPage = $(PREVIOUS_PAGE_SELECTOR, opts.paginationSelector); // Next page button
+    ctx._pfPagination.totalItems = $(TOTAL_ITEMS_SELECTOR, opts.paginationSelector); // Total items
+    ctx._pfPagination.totalPages = $(TOTAL_PAGES_SELECTOR, opts.paginationSelector); // Total pages
```
Comment (jeff-phillips-18) [inline src/js/patternfly.dataTables.pfPagination.js:127]: While comments are good, having the comment simply restate the variable name only leads to them becoming outdated when things change. Feel free to ignore.

**Review by jeff-phillips-18 — APPROVED** [review]
👍

**Comment by dlabrecq** [comment]
Thanks, Dave. The select all checkbox is a separate bug, but I've added a fix for that.

**Pull request merged**